src/Agent.Sdk/Util/RepositoryUtil.cs

```diff
@@ -0,0 +1,164 @@
+            // Trim the rest of the string after the colon if it is empty or is all digits
+            if (lastColon >= 0 && (lastColon == endIndex || buffer.IsNumber(lastColon + 1, endIndex)))
+            {
+                return lastColon - 1;
+            }
+
+            return endIndex;
+        }
+
+        private static int TrimSlashesAndExtension(string buffer, int endIndex)
+        {
+            if (buffer == null || endIndex < 0 || endIndex >= buffer.Length)
+            {
+                return endIndex;
+            }
+
+            // skip ending slashes or whitespace
+            while (endIndex > 0 && (buffer[endIndex] == '/' || char.IsWhiteSpace(buffer[endIndex])))
+            {
+                endIndex--;
+            }
+
+            const string gitExtension = ".git";
+            int possiblExtStart = endIndex - gitExtension.Length + 1;
+            if (possiblExtStart >= 0 && gitExtension.Equals(buffer.Substring(possiblExtStart, gitExtension.Length), StringComparison.OrdinalIgnoreCase))
+            {
+                // We found the .git extension
+                endIndex -= gitExtension.Length;
+            }
+
+            // skip ending slashes or whitespace
+            while (endIndex > 0 && (buffer[endIndex] == '/' || char.IsWhiteSpace(buffer[endIndex])))
+            {
+                endIndex--;
+            }
+
+            return endIndex;
+        }
+
+        private static int SkipLastIndexOf(this string buffer, char charToSearchFor, int startIndex, int endIndex, out bool charFound)
+        {
+            int index = buffer.FinalIndexOf(charToSearchFor, startIndex, endIndex);
+            if (index >= 0 && index < endIndex)
+            {
+                // Start after the char we found
+                charFound = true;
+                return index + 1;
+            }
+
+            charFound = false;
+            return startIndex;
+        }
+
+        private static int FinalIndexOf(this string buffer, char charToSearchFor, int startIndex, int endIndex)
+        {
+            if (buffer == null || startIndex < 0 || endIndex < 0 || startIndex >= buffer.Length || endIndex >= buffer.Length)
+            {
+                return -1;
+            }
+
+            return buffer.LastIndexOf(charToSearchFor, endIndex, endIndex - startIndex + 1);
+        }
+
+        private static bool IsNumber(this string buffer, int startIndex, int endIndex)
+        {
+            bool digitsFound = false;
+
+            if (buffer == null || startIndex < 0 || endIndex < 0 || startIndex >= buffer.Length || endIndex >= buffer.Length)
+            {
+                return false;
+            }
+
+            for (int i = startIndex; i <= endIndex; i++)
+            {
+                if (!char.IsDigit(buffer[i]))
+                {
+                    return false;
+                }
+                else
+                {
+                    digitsFound = true;
```

Comment:
Do we need `digitsFound`? If we iterate each character and return false if it's not a digit then I think at the end of the function we can just return true.

Reply:
I wanted to handle the case when the for loop doesn't execute. In that case, no digits where found and so the result if false. I can probably get around that with a check at the top.